File: src/zoo.ts
import type { Animal } from "@/animal";

export class Zoo {
  private _name: string;
  private _animals: Animal[];

  constructor(name: string, animals: Animal[]) {
    this._name = name;
    this._animals = animals;
  }

  public add(animal: Animal): void {
    this._animals.push(animal);
  }

  public showAnimals(): void {
    this._animals.map((animal) => console.log(animal.toString()));
  }

  public toString(): string {
    return `Zoo[name=${this._name}, animals=${this._animals.length}]`;
  }
}

File: src/zoo.rs
use crate::animal::Animal;

pub struct Zoo {
    name: String,
    animals: Vec<Box<dyn Animal>>,
}

impl Zoo {
    pub fn new(name: String, animals: Vec<Box<dyn Animal>>) -> Self {
        Self { name, animals }
    }

    pub fn add(&mut self, animal: Box<dyn Animal>) {
        self.animals.push(animal);
    }

    pub fn show_animals(&self) {
        for animal in &self.animals {
            println!("{}", animal.to_string());
        }
    }

    pub fn to_string(&self) -> String {
        format!("Zoo[name={}, animals: {}]", self.name, self.animals.len())
    }
}
